pkg/chains/signing.go

```diff
@@ -58,21 +62,47 @@ type TaskRunSigner struct {
 }
 
 // IsSigned determines whether a TaskRun has already been signed.
-func IsSigned(tr *v1beta1.TaskRun) bool {
+func IsSigned(tr *v1beta1.TaskRun) (bool, string) {
+	const signed = "true"
 	signature, ok := tr.ObjectMeta.Annotations[ChainsAnnotation]
 	if !ok {
-		return false
+		return false, ""
 	}
-	return signature == "true"
+	return signature == signed, signature
+}
+
+// returns true if:
+// the signed annotation isn't there, the failed annotation isn't there, and retries<3
+func RetryAgain(tr *v1beta1.TaskRun) (bool, string) {
```

Comment:
This can just return `true/false`, we don't need to return the string of retries.

Instead, we can get the value of that string in `AddRetries` and then add to it there.

Reply:
Took care of it. Thanks